File: codingame/Shadows of the Knight - Episode 1/main.dart
import 'dart:io';
import 'dart:math';

String readLineSync() {
  String? s = stdin.readLineSync();
  return s == null ? '' : s;
}

void main() {
  List inputs;
  inputs = readLineSync().split(' ');
  int W = int.parse(inputs[0]);
  int H = int.parse(inputs[1]);
  int N = int.parse(readLineSync());
  inputs = readLineSync().split(' ');
  int X0 = int.parse(inputs[0]);
  int Y0 = int.parse(inputs[1]);

  List<int> possibleWidth = [for (var i = 0; i < W; ++i) i];
  List<int> possibleHeight = [for (var i = 0; i < H; ++i) i];
  int currentPosX = X0;
  int currentPosY = Y0;

  while (true) {
    String bombDir = readLineSync();

    if (bombDir.contains("L")) {
      possibleWidth =
          possibleWidth.sublist(0, possibleWidth.indexOf(currentPosX));
      currentPosX = possibleWidth[(possibleWidth.length / 2).floor()];
    } else if (bombDir.contains("R")) {
      possibleWidth =
          possibleWidth.sublist(possibleWidth.indexOf(currentPosX) + 1);
      currentPosX = possibleWidth[(possibleWidth.length / 2).floor()];
    }

    if (bombDir.contains("U")) {
      possibleHeight =
          possibleHeight.sublist(0, possibleHeight.indexOf(currentPosY));
      currentPosY = possibleHeight[(possibleHeight.length / 2).floor()];
    } else if (bombDir.contains("D")) {
      possibleHeight =
          possibleHeight.sublist(possibleHeight.indexOf(currentPosY) + 1);
      currentPosY = possibleHeight[(possibleHeight.length / 2).floor()];
    }

    print('$currentPosX $currentPosY');
  }
}

File: codingame/Shadows of the Knight - Episode 1/main.js
var inputs = readline().split(' ');
const W = parseInt(inputs[0]);
const H = parseInt(inputs[1]);
const N = parseInt(readline());
var inputs = readline().split(' ');
const X0 = parseInt(inputs[0]);
const Y0 = parseInt(inputs[1]);

var possibleWidth = [...Array(W).keys()];
var possibleHeight = [...Array(H).keys()];
var currentPosX = X0;
var currentPosY = Y0;

while (true) {
    const bombDir = readline();

    if (bombDir.includes("L")) {
        possibleWidth = possibleWidth.slice(0, possibleWidth.indexOf(currentPosX));
        currentPosX = possibleWidth[Math.floor(possibleWidth.length/2)];
    }
    else if (bombDir.includes("R")) {
        possibleWidth = possibleWidth.slice(possibleWidth.indexOf(currentPosX)+1);
        currentPosX = possibleWidth[Math.floor(possibleWidth.length/2)];
    }

    if (bombDir.includes("U")) {
        possibleHeight = possibleHeight.slice(0, possibleHeight.indexOf(currentPosY));
        currentPosY = possibleHeight[Math.floor(possibleHeight.length/2)];
    }
    else if (bombDir.includes("D")) {
        possibleHeight = possibleHeight.slice(possibleHeight.indexOf(currentPosY)+1);
        currentPosY = possibleHeight[Math.floor(possibleHeight.length/2)];
    }

    console.log(`${currentPosX} ${currentPosY}`);
}
